src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
+		boolean result3 = checkLuhnalgorithmCorrectness(number);
+		
+		return result1 && result2 && result3;
+	}
+	
+	private boolean checkLengthCorrectness(String number, ArrayList<Integer> chars){
```

Comment:
Consider changing ArarayList<Integer> to Set<Integer> this would significantly reduce code  with much impact on peforamnce